misk/src/main/kotlin/misk/web/actions/NotFoundAction.kt

```diff
@@ -1,29 +1,41 @@
 package misk.web.actions
 
 import misk.logging.getLogger
+import misk.scope.ActionScoped
 import misk.security.authz.Unauthenticated
 import misk.web.Get
+import misk.web.HttpCall
 import misk.web.PathParam
 import misk.web.Post
 import misk.web.RequestContentType
 import misk.web.Response
 import misk.web.ResponseBody
 import misk.web.ResponseContentType
+import misk.web.jetty.WebActionsServlet
 import misk.web.mediatype.MediaTypes
 import misk.web.toResponseBody
 import okhttp3.Headers.Companion.headersOf
 import javax.inject.Inject
+import javax.inject.Provider
 import javax.inject.Singleton
 
 @Singleton
-class NotFoundAction @Inject constructor() : WebAction {
+class NotFoundAction @Inject constructor(
+  @JvmSuppressWildcards private val clientHttpCall: ActionScoped<HttpCall>
+) : WebAction {
+  @Inject internal lateinit var servletProvider: Provider<WebActionsServlet>
+
   @Get("/{path:.*}")
   @Post("/{path:.*}")
   @RequestContentType(MediaTypes.ALL)
   @ResponseContentType(MediaTypes.ALL)
   @Unauthenticated
   fun notFound(@PathParam path: String): Response<ResponseBody> {
-    return response(path)
+    return Response(
+        body = notFoundMessage(path, clientHttpCall.get()).toResponseBody(),
+        headers = headersOf("Content-Type", MediaTypes.TEXT_PLAIN_UTF8),
+        statusCode = 404
```

Comment:
Was NotFoundAction not returning a `404` before?

Reply:
nah, it was on the old line 37

Reply:
It was, but it was doing so with the `response()` method of its Companion, which isn't capable of providing the contextful response.